hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/HederaEvmAccount.java

```diff
@@ -67,8 +68,11 @@ public interface HederaEvmAccount extends MutableAccount {
      * cache of {@link org.hyperledger.besu.evm.Code} wrappers around raw bytecode returned by
      * {@link Account#getCode()}.
      *
+     * @param functionSelector the function selector to use when fetching the code.  If more than 4 bytes for the
+     *                         function selector is passed in, only the first 4 bytes will be used.
+     *                         Only relevant for regular accounts.
      * @return the EVM code for this account
      */
     @NonNull
-    Code getEvmCode();
+    Code getEvmCode(@NonNull final Bytes functionSelector);
```

Comment:
We don't need this argument any more, do we?

Reply:
we might if we want to restrict `TokenEvmAccount` and `ProxyEvmAccount` to return the proxy contract byte code only if the function selector matches the expected functions on tokens/accounts.